agent/stats/queue.go

```diff
@@ -92,15 +92,15 @@ func (queue *Queue) add(rawStat *ContainerStats) {
 		// % utilization can be calculated only when queue is non-empty.
 		lastStat := queue.buffer[queueLength-1]
 		timeSinceLastStat := float32(rawStat.timestamp.Sub(lastStat.Timestamp).Nanoseconds())
-		if timeSinceLastStat > 0 {
+		if timeSinceLastStat <= 0 {
+			// if we got a duplicate timestamp, set cpu percentage to the same value as the previous stat
+			seelog.Errorf("Received a docker stat object with duplicate timestamp")
+			stat.CPUUsagePerc = lastStat.CPUUsagePerc
+		} else {
```

Comment:
Just wonder why we are not directly using raw cpuUsage and do this calculation instead?

Reply:
because raw cpu usage is measure in "cpu time", for which the units is nanoseconds and is just a counter of the amount of time the CPUs were utilized. Most people want this converted to "usage percent" because it's more usable, but you need two measure of cpu time to figure out the usage percent.